yt/geometry/selection_routines.pyx

```diff
@@ -1232,9 +1233,12 @@ cdef class SliceSelector(SelectorObject):
                 this_level = 1
             for i in range(3):
                 if i == self.axis:
-                    ind[i][0] = \
-                        <int> ((self.coord - (gobj.LeftEdge[i]).to_ndarray()) /
-                               gobj.dds[i])
+                    icoord = <np.uint8_t>(
+                        (self.coord - gobj.LeftEdge.d[i])/gobj.dds[i])
+                    # clip coordinate to avoid seg fault below if we're
+                    # exactly at a grid boundary
+                    icoord = min(icoord, gobj.ActiveDimensions[i]-1)
+                    ind[i][0] = max(icoord, 0)
```

Comment:
Is there a reason you don't use the clip function defined in `yt/utilities/lib/fp_utils.pyx`?

Reply:
Nope, I can use that, that's probably faster.